sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -78,19 +90,24 @@ abstract class StatePartitionReaderBase(
   extends PartitionReader[InternalRow] with Logging {
   // Used primarily as a placeholder for the value schema in the context of
   // state variables used within the transformWithState operator.
-  private val schemaForValueRow: StructType =
+  // Also used as a placeholder for both key and value schema for
+  // StatePartitionAllColumnFamiliesReader
+  private val placeholderSchema: StructType =
```

Comment:
we shouldn't do this. The keySchema and valueSchema vals below are just used to open the provider and are different from the key/value schema returned by the reader. Lets use the actual key/value schema here, to make sure that we are correctly opening the provider. I mentioned this in my previous review: https://github.com/apache/spark/pull/53104#discussion_r2539759286

Then you wouldn't need the format validation config thing you are doing